src/partition/ReceivedPartition.cpp

```diff
@@ -585,7 +585,7 @@ void ReceivedPartition::createOwnerInformation()
     }
 
     // Send information back to slaves
-    for (int rank = 1; rank < utils::MasterSlave::getSize(); rank++) {
+    for (int rank : utils::MasterSlave::slaves()) {
```

Comment:
Maybe it would be better to find some consensus on how to name the iterator. It is not really intuitive that the loop above starts from zero whereas the loop here starts from one. Maybe we should even rename `::ranks()` into `::allRanks()` in order to emphasize that the master rank is included.

Reply:
> It is not really intuitive that the loop above starts from zero whereas the loop here starts from one.

That is the reason I implemented `slaves()` and `ranks()`.

> Maybe we should even rename ::ranks() into ::allRanks() in order to emphasize that the master rank is included.

I was wondering about this too. `allRanks()` is good as it makes both the range  and the type explicit.
To keep things consistent I would then rename `slaves()` to `slaveRanks()`.

Reply:
> I was wondering about this too. allRanks() is good as it makes both the range and the type explicit.
To keep things consistent I would then rename slaves() to slaveRanks().

Agree :ok_hand: